doc/README-zh-cn.md

```diff
@@ -163,95 +163,96 @@ var storage = multer.diskStorage({
+`destination` 是用来确定上传的文件应该存储在哪个文件夹中。也可以提供一个 `string` (例如 `'/tmp/uploads'`)。如果没有设置 `destination`，则使用操作默认的临时文件夹
 
-**ע��:** ������ṩ�� `destination` ��һ��������Ҫ���𴴽��ļ��С����ṩһ���ַ�multer ��ȷ������ļ������㴴���ġ�
+**注意:** 如果你提供的 `destination` 是一个函数，你需要负责创建文件夹。当提供一个字符串，multer 将确保这个文件夹是你创建的。
 
-`filename` ����ȷ���ļ����е��ļ����ȷ���� ���û������ `filename`��ÿ���ļ�������Ϊһ������ļ�������û���չ��� 
+`filename` 用于确定文件夹中的文件名的确定。 如果没有设置 `filename`，每个文件将设置为一个随机文件名，并且是没有扩展名的 
 
-**ע��:** Multer ����Ϊ������κ��չ����ĳ���Ӧ�÷���һ��������ļ���
+**注意:** Multer 不会为你添加任何扩展名，你的程序应该返回一个完整的文件名。
 
-ÿ�������������� (`req`) ��һЩ��������ļ�����Ϣ (`file`) ��������ľ��
+每个函数都传递了两个请求 (`req`) 和一些关于这个文件的信息 (`file`) 有助于你的决定。
 
-ע�� `req.body` ���ܻ�û����ȫ��䣬��ȡ������ͻ��˷����ֶκ��ļ����������˳��
+注意 `req.body` 可能还没有完全填充，这取决于向客户端发送字段和文件到服务器的顺序。
 
 #### `MemoryStorage`
 
-�ڴ�洢���潫�ļ��洢���ڴ��е� `Buffer` ������û���κ�ѡ��
+内存存储引擎将文件存储在内存中的 `Buffer` 对象，它没有任何选项
 
 ```javascript
 var storage = multer.memoryStorage()
 var upload = multer({ storage: storage })
 ```
 
-��ʹ���ڴ�洢���棬�ļ���Ϣ����һ�� `buffer` �ֶΣ������������ļ���ݡ�
+当使用内存存储引擎，文件信息将包含一个 `buffer` 字段，里面包含了整个文件数据。
 
-**����**: ����ʹ���ڴ�洢���ϴ��ǳ�����ļ������߷ǳ����С�ļ����ᵼ�����Ӧ�ó����ڴ����
+**警告**: 当你使用内存存储，上传非常大的文件，或者非常多的小文件，会导致你的应用程序内存溢出
 
 ### `limits`
-һ�����ָ��һЩ��ݴ�С�����ơ�Multer ͨ��������ʹ�� busboy����ϸ�����Կ����� [busboy's page](https://github.com/mscdex/busboy#busboy-methods) �ҵ���
+一个对象，指定一些数据大小的限制。Multer 通过这个对象使用 busboy，详细的特性可以在 [busboy's page](https://github.com/mscdex/busboy#busboy-methods) 找到。
 
-����ʹ��������Щ:
+可以使用下面这些:
 
 Key | Description | Default
 --- | --- | ---
-`fieldNameSize` | field ������󳤶� | 100 bytes
-`fieldSize` | field ֵ����󳤶�  | 1MB
-`fields` | ���ļ� field �������� | ����
-`fileSize` | �� multipart ���У��ļ���󳤶� (�ֽڵ�λ) | ����
-`files` | �� multipart ���У��ļ������� | ����
-`parts` | �� multipart ���У�part �����������(fields + files) | ����
-`headerPairs` | �� multipart ���У���ֵ��������� | 2000
+`fieldNameSize` | field 名字最大长度 | 100 bytes
+`fieldSize` | field 值的最大长度  | 1MB
+`fields` | 非文件 field 的最大数量 | 无限
+`fileSize` | 在 multipart 表单中，文件最大长度 (字节单位) | 无限
+`files` | 在 multipart 表单中，文件最大数量 | 无限
+`parts` | 在 multipart 表单中，part 传输的最大数量(fields + files) | 无限
+`headerPairs` | 在 multipart 表单中，键值对最大组数 | 2000
 
-���� limits ���԰������վ�����ܾܾ���� (DoS) ����
+设置 limits 可以帮助保护你的站点免受拒绝服务 (DoS) 攻击。
 
 ### `fileFilter`
-����һ���������ʲô�ļ������ϴ��Լ�ʲô�ļ�Ӧ����������Ӧ�ÿ����������
+设置一个函数来控制什么文件可以上传以及什么文件应该跳过，这个函数应该看起来像这样：
 
 ```javascript
 function fileFilter (req, file, cb) {
 
-  // �����Ӧ�õ��� `cb` ��booleanֵ�
-  // ָʾ�Ƿ�Ӧ���ܸ��ļ�
+  // 这个函数应该调用 `cb` 用boolean值来
+  // 指示是否应接受该文件
+  // cb 的第一个参数应传入一个Error或是null
 
-  // �ܾ�����ļ���ʹ��`false`��������:
+  // 拒绝这个文件，使用`false`，像这样:
   cb(null, false)
 
-  // ��������ļ���ʹ��`true`��������:
+  // 接受这个文件，使用`true`，像这样:
   cb(null, true)
 
-  // ��������⣬���������������һ�����:
-  cb(new Error('I don\'t have a clue!'))
+  // 如果有问题，你可以总是这样发送一个错误:
+  cb(new Error('I don\'t have a clue!'), false)
```

Comment:
When an error is passed, the second parameter won't be looked at, so this shouldn't be necessary 🤔

Did this not work for you without the `false` parameter?